libc/src/stdio/scanf_core/reader.h

```diff
@@ -9,15 +9,73 @@
 #ifndef LLVM_LIBC_SRC_STDIO_SCANF_CORE_READER_H
 #define LLVM_LIBC_SRC_STDIO_SCANF_CORE_READER_H
 
+// has the macro subsitution failure
+// #ifdef LIBC_FULL_BUILD
+#include "src/__support/File/file.h"
```

Comment:
So, on macOS (locally) (and I'm pretty sure this will show up in the CI as well),

This file ends up including the system `stdio.h`

Which has macros defined like `clearerr_unlocked` which are leaking into this TU and causing syntax errors at:

https://github.com/llvm/llvm-project/blob/f66a5e220cbc2650a5843db854d0734d2aaa030f/libc/src/__support/File/file.h#L260

So stuck here,

one way could be changing the names of these methods in `__support/File/file` but that API may be in use by others, so changing it may not be an option

Reply:
See https://github.com/llvm/llvm-project/pull/121215#discussion_r1921280522 for fix